index.js

```diff
@@ -12,21 +12,21 @@ function fastifyMysql (fastify, options, next) {
   _createConnection(options, (err, db) => {
     if (err) return next(err)
 
-    const client = (connectionType !== 'connection') ? db.pool : db.connection
+    const client = connectionType !== 'connection' ? db.pool : db.connection
 
     if (name) {
       if (!fastify.mysql) {
         fastify.decorate('mysql', {})
       }
+
       if (fastify.mysql[name]) {
-        next(new Error('fastify.mysql.' + name + 'has already registered'))
-        return
```

Comment:
attention, without the return statement we are calling the `next()` function twice with unpredictable behaviours

Reply:
Oups that was a typo, it should return the `next` callback ^^
I must check the test as the weird timeout I experienced on the master branch after updating the dependencies is occurring again ^^

Reply:
Ok there is definitely something going wrong with the plugin,
when it throws an error there is a weird timeout and it seems that the connection is not properly closed (seems like the onClose hook is not fired).

Will dig much more into this tomorrow.

Reply:
I'll give you a hand later today if you need 💪🏻

Reply:
Hello ^^,
Problem solved :d
As I guessed it was just a matter of when the `onClose` hook was registered and triggered.